druid/src/window.rs

```diff
@@ -180,6 +180,19 @@ impl<T: Data> Window<T> {
             );
         }
 
+        /*
+         * If a Widget is newly created it would not have an initial layout until
+         * paint time whichs bad because we might send more events down the tree
+         * before then. This is bad because a widget is not supposed to recieve a
+         * WidgetAdded event and then have layout before recieving any other events.
+         * Avoid this by performing a layout now that it has recieved WidgetAdded
+         * but before any other events could be sent.
+         */
+        if widget_state.needs_layout {
+            self.invalidate_and_finalize();
+            self.layout(queue, data, env);
+        }
+
```

Comment:
I figured it would be simpler to always perform this layout as opposed to placing it in the `if widget_state.children_changed` block directly above. I am not confident about this